io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -724,4 +732,30 @@ private List<String> writeToFile(
 		return filesWritten;
 	}
 
+	/**
+	 * 
+	 * @param filePath Path of the file to be written with just one line of the
+	 *            encoding
+	 * @return File that has been created filled with the text
+	 * @throws IOException exception in any.
+	 */
+	public File writeCodePageFile(String filePath) throws IOException {
+		File file = new File(filePath);
+		FileWriter fr = new FileWriter(file);
+		try {
+			fr.write(getCharset() != null ? getCharset().toString()
+					: getDefaultCharset().toString());
+		} catch (IOException e) {
+			throw e;
+		} finally {
+			try {
+				fr.close();
+			} catch (IOException e) {
+				throw new IOException("An error occurred while creating/writing the file: "
```

Comment:
This exception will be thrown when closing the file so the text should be modified. 

I think this particular error message should move to the above catch block (you may also mention `cpg` somewhere in the text to be more specific for the origin of the exception).